dns: default to verbatim=true in dns.lookup()

bnoordhuis opened the pull request:
Switch the default from false (reorder the result so that IPv4 addresses
come before IPv6 addresses) to true (return them exactly as the resolver
sent them to us.)

This is intended to be released in Node.js 11.

vsemozhetbyt commented on doc/api/dns.md:
```diff
@@ -144,10 +144,7 @@ changes:
-    **Default:** currently `false` (addresses are reordered) but this is
-    expected to change in the not too distant future.
-    New code should use `{ verbatim: true }`.
+    IPv4 addresses are placed before IPv6 addresses. **Default:** `true`
```
Nit: a period after `true` for consistency with the previous option.

BridgeAR commented:
There are lots of related CI failures.

bnoordhuis commented:
I kinda expected that, I was pretty sure the test suite would be full of assumptions about localhost.  I fixed them, PTAL.  I also removed `common.localhostIPv4`, see the commit log for rationale.

https://ci.nodejs.org/job/node-test-pull-request/14958/

jasnell commented:
Needs a rebase and another CI run after.

BridgeAR commented:
CI https://ci.nodejs.org/job/node-test-pull-request/15149/

bnoordhuis commented:
Thanks, Ruben.  Looks like there are still some tests to fix up...

refack commented:
Can one of the admins verify this patch?

jasnell commented:
ping @bnoordhuis

jasnell commented:
ping @bnoordhuis

Trott commented:
@bnoordhuis I've rebased and resolved conflicts. A lot of tests still fail. Should I force-push to your branch and then whoever wants to try to finish it (you, me, someone else) can? Or would you rather resolve the conflicts yourself, etc.?

bnoordhuis commented:
@Trott You're welcome to force-push. I do intend to pick this up again whenever I have some spare time.

BridgeAR commented:
Closing due to long inactivity. @bnoordhuis please reopen / open a new PR in case you want to continue working on this.

bnoordhuis commented:
Continues in #31567.
